clippy_lints/src/methods/redundant_repeat_take.rs

```diff
@@ -0,0 +1,32 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet;
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind, QPath};
+use rustc_lint::LateContext;
+
+use super::REDUNDANT_REPEAT_TAKE;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, repeat_expr: &Expr<'_>, take_arg: &Expr<'_>) {
+    if !expr.span.from_expansion()
+        && let ExprKind::Call(inner, repeat_args) = repeat_expr.kind
+        && repeat_args.len() == 1
+        && let ExprKind::Path(ref qpath) = inner.kind
+        && let QPath::Resolved(_, path) = qpath
+        && path.segments.len() == 1
+        && path.segments[0].ident.name.as_str() == "repeat"
+    {
+        span_lint_and_sugg(
+            cx,
+            REDUNDANT_REPEAT_TAKE,
+            expr.span,
+            "redundant `repeat().take()`",
+            "consider using `repeat_n()` instead",
+            format!(
+                "std::iter::repeat_n({}, {})",
+                snippet(cx, repeat_args[0].span, ""),
+                snippet(cx, take_arg.span, "")
```

Comment:
Would be good to have a test case where the element is a macro call, e.g. `repeat(vec![1, 2]).take(2)`. I suspect this currently "doesn't work" (has macro expansion in the suggestion), so this may need to use `snippet_with_context` (with `expr.span.ctxt()` for the ctxt), but I have not tested it